components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,341 @@
+                                        new AuthProperty.AuthPropertyBuilder().name("password").value("admin").isConfidential(true).build()))
+                                .build())
+                        .build())
+                .build();
+    }
+
+    private Action getActionWithoutOptionals(String id){
+        return new Action.ActionResponseBuilder()
+                .id(id)
+                .type(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN)
+                .name("Pre Issue Access Token")
+                .status(Action.Status.ACTIVE)
+                .endpoint( new EndpointConfig.EndpointConfigBuilder()
+                        .uri("https://example.com")
+                        .authentication(new AuthType.AuthTypeBuilder()
+                                .type(AuthType.AuthenticationType.BASIC)
+                                .properties(Arrays.asList(
+                                        new AuthProperty.AuthPropertyBuilder().name("username").value("admin").isConfidential(true).build(),
+                                        new AuthProperty.AuthPropertyBuilder().name("password").value("admin").isConfidential(true).build()))
+                                .build())
+                        .build())
+                .build();
+    }
+
+    private void loadMockDatabaseUtil() {
+        try {
+            identityDatabaseUtil.when(() -> IdentityDatabaseUtil.getDBConnection(anyBoolean()))
+                    .thenReturn(getConnection());
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Connection getConnection() throws SQLException {
+
+        if (dataSourceMap.get(DB_NAME) != null) {
+            return dataSourceMap.get(DB_NAME).getConnection();
+        }
+        throw new RuntimeException("No datasource initiated for database: " + ActionManagementDAOImplTest.DB_NAME);
+    }
+
+    private void initiateH2Database(String scriptPath) throws Exception {
+
+        BasicDataSource dataSource = new BasicDataSource();
+        dataSource.setDriverClassName("org.h2.Driver");
+        dataSource.setUsername("username");
+        dataSource.setPassword("password");
+        dataSource.setUrl("jdbc:h2:mem:test" + DB_NAME);
+        dataSource.setTestOnBorrow(true);
+        dataSource.setValidationQuery("select 1");
+        try (Connection connection = dataSource.getConnection()) {
+            connection.createStatement().executeUpdate("RUNSCRIPT FROM '" + scriptPath + "'");
+        }
+        dataSourceMap.put(DB_NAME, dataSource);
+    }
+
+    private static String getFilePath() {
+
+        if (StringUtils.isNotBlank("h2.sql")) {
+            return Paths.get(System.getProperty("user.dir"), "src", "test", "resources", "dbscripts", "h2.sql")
+                    .toString();
+        }
+        throw new IllegalArgumentException("DB Script file name cannot be empty.");
+    }
+
+    public static void closeH2Database() throws Exception {
```

Comment:
Check whether we can specify the exact exception type thrown by this method.
Check the exception thrown by the datasource.close() method.

Reply:
datasource.close() method throws SQLException. I edited with it . Is that sufficient ?